packages/app/src/app/pages/Sandbox/Editor/Content/Preview/index.tsx

```diff
@@ -9,198 +11,249 @@ import getTemplate from '@codesandbox/common/lib/templates';
    * a better design if we want to use it for more though.
    */
-  getOverlayMessage = () => {
-    const {
-      containerStatus,
-      error,
-      hasUnrecoverableError,
-    } = this.props.store.server;
-
-    if (containerStatus === 'hibernated') {
-      return 'The container has been hibernated because of inactivity, you can start it by refreshing the browser.';
-    }
+  let overlayMessage: string;
+  const { containerStatus, error, hasUnrecoverableError } = server;
 
-    if (containerStatus === 'stopped') {
-      return 'Restarting the sandbox...';
-    }
+  if (containerStatus === 'hibernated') {
+    overlayMessage =
+      'The container has been hibernated because of inactivity, you can start it by refreshing the browser.';
+  }
 
-    if (error && hasUnrecoverableError) {
-      return 'A sandbox error occurred, you can refresh the page to restart the container.';
-    }
+  if (containerStatus === 'stopped') {
+    overlayMessage = 'Restarting the sandbox...';
+  }
 
-    return undefined;
-  };
-
-  render() {
-    const { store, signals, options } = this.props;
-
-    const completelyHidden = !store.editor.previewWindowVisible;
-
-    return this.state.running ? (
-      <BasePreview
-        onInitialized={this.onPreviewInitialized}
-        sandbox={store.editor.currentSandbox}
-        currentModule={store.editor.currentModule}
-        settings={store.preferences.settings}
-        initialPath={store.editor.initialPath}
-        url={options.url}
-        isInProjectView={store.editor.isInProjectView}
-        onClearErrors={() => signals.editor.errorsCleared()}
-        onAction={action => signals.editor.previewActionReceived({ action })}
-        hide={this.props.hidden}
-        noPreview={completelyHidden}
-        onToggleProjectView={() => signals.editor.projectViewToggled()}
-        isResizing={store.editor.isResizing}
-        overlayMessage={this.getOverlayMessage()}
-      />
-    ) : (
-      <RunOnClick
-        onClick={() => {
-          this.setState({ running: true });
-        }}
-      />
-    );
+  if (error && hasUnrecoverableError) {
+    overlayMessage =
+      'A sandbox error occurred, you can refresh the page to restart the container.';
   }
-}
 
-export const Preview = inject('signals', 'store')(observer(PreviewComponent));
+  return running ? (
+    <BasePreview
+      onInitialized={handlePreviewInitialized}
+      sandbox={currentSandbox}
+      currentModule={currentModule}
+      settings={settings}
+      initialPath={initialPath}
+      url={url}
+      isInProjectView={isInProjectView}
+      onClearErrors={handleErrorsCleared}
+      onAction={handlePreviewAction}
+      hide={hidden}
+      noPreview={completelyHidden}
+      onToggleProjectView={handleProjectViewToggled}
+      isResizing={isResizing}
+      overlayMessage={overlayMessage}
+    />
+  ) : (
+    <RunOnClick onClick={handleRunOnClick} />
+  );
+};
+
+export const Preview = PreviewComponent;
```

Comment:
```suggestion

```

Let's just remove this entirely, no need to rename the component and export this way.

Reply:
done